packages/core/factories/src/useProductFactory.ts

```diff
@@ -19,16 +19,16 @@ export function useProductFactory<PRODUCT, PRODUCT_SEARCH_PARAMS>(
 ) {
   return function useProduct(cacheId: string): UseProduct<PRODUCT> {
     const { initialState, saveToInitialState } = useSSR(cacheId);
-    const products: Ref<PRODUCT[]> = ref(initialState || []);
-    const totalProducts = ref(0);
+    const products: Ref<PRODUCT[]> = ref(initialState ? initialState.data : []);
+    const totalProducts: Ref<number> = ref(initialState ? initialState.total : 0);
```

Comment:
Just a little suggestion here :) 

```suggestion
    const products: Ref<PRODUCT[]> = ref(initialState?.data || []);
    const totalProducts: Ref<number> = ref(initialState?.total || 0);
```